editoast/src/views/timetable/similar_trains/graph.rs

```diff
@@ -153,13 +153,13 @@ impl MatchingState {
     pub(super) fn new(segment: Segment, graph: Graph) -> Self {
```

Comment:
```suggestion
    pub(super) fn try_new(segment: Segment, graph: Graph) -> Result<Self, graph::Waypoint> {
```

There is no sense in having a matching object that's unable to perform a single match. We should refuse to provide the matching handle and let the caller deal with the situation.

Reply:
I used `Option` instead of `Result` to make it easier, returning a matching object or nothing, so the caller can handle it.